src/scenes/Root/CommentsBar/CommentItem/CommentItem.tsx

```diff
@@ -0,0 +1,179 @@
+import { useMutation } from '@apollo/client';
+import { MoreVert } from '@mui/icons-material';
+import { Box, IconButton, MenuProps, Paper, Typography } from '@mui/material';
+import Blocks from 'editorjs-blocks-react-renderer';
+import { useState } from 'react';
+import { removeItemFromList } from '~/api';
+import { useDateTimeFormatter } from '~/components/Formatters';
+import { CommentItemMenu } from '../CommentItemMenu';
+import { CommentReply } from '../CommentReply';
+import {
+  CommentPropsFragment,
+  CommentThreadPropsFragment,
+  DeleteCommentDocument,
+} from '../CommentsBar.graphql';
+import { useCommentsContext } from '../CommentsBarContext';
+
+export interface CommentProps {
+  comment: CommentPropsFragment;
+  repliesCount?: number;
+  isChild?: boolean;
+  resourceId: string;
+  isExpanded?: boolean;
+  parent: CommentThreadPropsFragment;
+  handleDeleteComment?: (comment: CommentPropsFragment) => Promise<void> | void;
+  handleEditComment?: (comment: CommentPropsFragment) => Promise<void> | void;
+}
+
+export const CommentItem = ({
+  comment,
+  repliesCount = 0,
+  isChild,
+  handleDeleteComment,
+  isExpanded,
+  parent,
+  resourceId,
+}: CommentProps) => {
+  const { toggleThreadComments } = useCommentsContext();
+  const [actionsAnchor, setActionsAnchor] = useState<MenuProps['anchorEl']>();
+  const [isEditing, setIsEditing] = useState(false);
+
+  const [deleteCommentMutation] = useMutation(DeleteCommentDocument, {});
+
+  const dateTimeFormatter = useDateTimeFormatter();
+  const formattedDateString = dateTimeFormatter(comment.modifiedAt);
+
+  const deleteComment = async (commentId: string) => {
+    const { data } = await deleteCommentMutation({
+      variables: {
+        id: commentId,
+      },
+      update: removeItemFromList({
+        listId: [parent, 'comments'],
+        item: comment,
+      }),
+    });
+
+    if (data?.deleteComment) {
```

Comment:
I believe the mutation function call throws on error (with the default error policy at least), so there's no need to check for success here.